### PR TITLE
enhance: support recall estimation

### PR DESCRIPTION
**Pull request by chasingegg**
issue: #37899 
Only `search` api will be supported

### CODE REVIEW

**Comment by chasingegg** [comment]
rerun go-sdk

**Comment by chasingegg** [comment]
rerun go-sdk

**Comment by chasingegg** [comment]
rerun cpp-unit-test

**Comment by chasingegg** [comment]
rerun go-sdk

**Comment by chasingegg** [comment]
rerun go-sdk

**Comment by chasingegg** [comment]
rerun go-sdk

**Comment by chasingegg** [comment]
rerun go-sdk

File: pkg/metrics/proxy_metrics.go [hunk 1]
```diff
@@ -408,6 +408,15 @@ var (
 			Name:      "retry_search_result_insufficient_cnt",
 			Help:      "counter of retry search which does not have enough results",
 		}, []string{nodeIDLabelName, queryTypeLabelName, collectionName})
+
+	// ProxyRecallSearchCount records the counter that users issue recall evaluation requests, which are cpu-intensive
+	ProxyRecallSearchCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.ProxyRole,
+			Name:      "recall_serch_cnt",
```
Comment (czs007) [inline pkg/metrics/proxy_metrics.go:417]: typo of serch

Comment (chasingegg) [reply pkg/metrics/proxy_metrics.go]: fixed

**Pull request merged**